modules/config/src/main/scala/scala/cli/config/Keys.scala

```diff
@@ -54,4 +60,104 @@ object Keys {
 
   lazy val map: Map[String, Key[_]] = all.map(e => e.fullName -> e).toMap
 
+  private final case class RepositoryCredentialsAsJson(
+    host: String,
+    user: Option[String] = None,
+    password: Option[String] = None,
+    realm: Option[String] = None,
+    optional: Option[Boolean] = None,
+    matchHost: Option[Boolean] = None,
+    httpsOnly: Option[Boolean] = None,
+    passOnRedirect: Option[Boolean] = None
+  ) {
+    def credentials: RepositoryCredentials = {
+      var cred = RepositoryCredentials(host = host)
+      for (u <- user)
+        PasswordOption.parse(u) match {
+          case Left(error) =>
+            throw new Exception(
+              s"Malformed repository credentials user value (expected 'value:…', or 'file:/path', or 'env:ENV_VAR_NAME'): $error"
+            )
+          case Right(value) =>
+            cred = cred.copy(user = Some(value))
+        }
+      for (p <- password if p.nonEmpty)
+        PasswordOption.parse(p) match {
+          case Left(error) =>
+            throw new Exception(
+              s"Malformed repository credentials password value (expected 'value:…', or 'file:/path', or 'env:ENV_VAR_NAME'): $error"
+            )
+          case Right(value) =>
+            cred = cred.copy(password = Some(value))
+        }
+      for (r <- realm)
+        cred = cred.copy(realm = Some(r))
+      for (opt <- optional)
+        cred = cred.copy(optional = Some(opt))
+      for (v <- matchHost)
+        cred = cred.copy(matchHost = Some(v))
+      for (v <- httpsOnly)
+        cred = cred.copy(httpsOnly = Some(v))
+      for (v <- passOnRedirect)
+        cred = cred.copy(passOnRedirect = Some(v))
+      cred
```

Comment:
Maybe it would be good enough to create the `RepositoryCredentials` at the end instead of using `var`

Reply:
Good point, should be fixed.

Reply:
Good point, should be fixed.